weaviate/graphql/search_properties.go

```diff
@@ -0,0 +1,58 @@
+package graphql
+
+import (
+	pb "github.com/weaviate/weaviate/grpc/generated/protocol/v1"
+)
+
+type Properties struct {
+	withProperties []string
+	withReferences []*Reference
+}
+
+func (p *Properties) WithProperties(properties ...string) *Properties {
+	p.withProperties = properties
+	return p
+}
+
+func (p *Properties) WithReferences(references ...*Reference) *Properties {
+	p.withReferences = references
+	return p
+}
+
+func (p *Properties) togrpc() *pb.PropertiesRequest {
+	props := &pb.PropertiesRequest{
+		NonRefProperties:          p.withProperties,
+		ReturnAllNonrefProperties: len(p.withProperties) == 0,
+	}
+	if len(p.withReferences) > 0 {
+		refProperties := make([]*pb.RefPropertiesRequest, len(p.withReferences))
+		for i := range p.withReferences {
+			refProperties[i] = p.withReferences[i].togrpc()
+		}
+		props.RefProperties = refProperties
+	}
+	// TODO: add support for json object properties
```

Comment:
suggestion: could we replace this TODO with an equivalent docstring explaining that "properties of type object/object[] are marshaled into `NonRefProperties` because the client doesn't know what the corresponding database types of each property is"

Reply:
I have changed the comment, can you check if it's OK now?

Reply:
yes, that'll do just fine, thanks!